docs/src/pages/components/text-fields/text-fields.md

```diff
@@ -146,6 +146,22 @@ or
 The floating label is absolutely positioned, it won't impact the layout of the page.
 You need to make sure that the input is larger than the label to display correctly.
 
+### Number inputs allowing non-numeric characters
+
+By specification, inputs of type "number" allow the character 'e' (and other characters such as '+', '-', '.') to be entered.
+
+```jsx
+<TextField inputProps={{ type: 'number' }} />
+```
+
+If you are looking to validate that the input contains only numeric characters, consider using a field of type "text" with the _pattern_ attribute:
+
+```jsx
+<TextField inputProps={{ inputmode: 'numeric', pattern: '[0-9]*' }} />
+```
+
+If you are looking to completely prevent the user from entering non-numeric characters, one approach would be to use a library such as [react-number-format](https://github.com/s-yadav/react-number-format). See [Integration with 3rd party input libraries](#integration-with-3rd-party-input-libraries) below for a demo.
+
```

Comment:
What do you think of this alternative version?

```suggestion
### type="number"

The usage of `<input type="number">` can create potentially usability issues:

- Users can insert the letter "e"
- Users can accidentally increase or decrease the number using the scroll wheel
- [and more](https://technology.blog.gov.uk/2020/02/24/why-the-gov-uk-design-system-team-changed-the-input-type-for-numbers/)

If you are only looking to validate that the input contains numeric characters, consider using a field of type "text" with the _pattern_ attribute:

```jsx
<TextField inputProps={{ inputmode: 'numeric', pattern: '[0-9]*' }} />
```

In the future, we might provide an [input number component](https://github.com/mui-org/material-ui/issues/19154).
```

Reply:
@oliviertassinari That looks good, I was just now working on an updated version as well, see here: https://gist.github.com/IwalkAlone/83e16f4c22e8e09f55ce49d24889973f
I would include the issue link at the end as well, maybe a link to the roadmap too since Number Input is on there?

Let me know if you prefer your version or the one in the gist and I can make the update

Reply:
A couple of thoughts on the gist:

- "poor" seems too strong for the accessibility support. The impact seems limited
- we don't recommand the pattern approach as a  replacement for type number, but as a viable alternative if validation is the main concern.
- I don't think that we should link the roadmap. The related issue is more appropriate.

Reply:
@oliviertassinari I've committed an updated version based on your comments. Thanks for the feedback 👍